src/Map/MapRasterizer_P.cpp

```diff
@@ -441,43 +441,25 @@ void OsmAnd::MapRasterizer_P::rasterizePolygon(
         canvas.drawPath(path, paint);
 }
 
-void OsmAnd::MapRasterizer_P::rasterizePolyline(
+bool OsmAnd::MapRasterizer_P::calcPathByTrajectory(
     const Context& context,
-    SkCanvas& canvas,
-    const std::shared_ptr<const MapPrimitiviser::Primitive>& primitive,
-    bool drawOnlyShadow)
+    const QVector<PointI>& points31,
+    SkPath& path,
+    float offset)
 {
-    const auto& points31 = primitive->sourceObject->points31;
-    const auto& area31 = context.area31;
-    const auto& env = context.env;
-
-    assert(points31.size() >= 2);
-
-    SkPaint paint = _defaultPaint;
-    if (!updatePaint(context, paint, primitive->evaluationResult, PaintValuesSet::Layer_1, false))
-        return;
-
-    bool ok;
+    int dir = offset < 0 ? -1: 1;
+    offset = abs(offset);
+    bool shift = static_cast<int>(offset) != 0 ? true : false;
 
-    ColorARGB shadowColor;
-    ok = primitive->evaluationResult.getIntegerValue(env->styleBuiltinValueDefs->id_OUTPUT_SHADOW_COLOR, shadowColor.argb);
-    if (!ok || shadowColor == ColorARGB::fromSkColor(SK_ColorTRANSPARENT))
-        shadowColor = context.shadowColor;
-
-    float shadowRadius;
-    ok = primitive->evaluationResult.getFloatValue(env->styleBuiltinValueDefs->id_OUTPUT_SHADOW_RADIUS, shadowRadius);
-    if (drawOnlyShadow && (!ok || shadowRadius <= 0.0f))
-        return;
-
-    SkPath path;
     int pointIdx = 0;
     bool intersect = false;
     int prevCross = 0;
-    PointF vertex;
+    OsmAnd::PointF vertex;
+    const auto& area31 = context.area31;
     const auto pointsCount = points31.size();
     auto pPoint = points31.constData();
-    PointF pVertex;
-    PointF tempVertex;
+    OsmAnd::PointF pVertex;
```

Comment:
It's just a habit. If it's not allowed according to project related code style - just let me know.

Reply:
It's useless in this case and increases amount of code that changes nothing.

Reply:
Yes, it's redundant, sure.
It just a habit - in case of huge project it can improve readability - IMHO.
Is it prohibited to use full datatype with name spaces? (according to code-style for instance) - or it's just notice?

Reply:
You're working 100% of the time inside OsmAnd namespace, I'd object it's improving anything in this specific case.